vlm/src/main/scala/geotrellis/contrib/vlm/geotiff/GeoTiffReprojectRasterSource.scala

```diff
@@ -22,33 +22,39 @@ import geotrellis.raster._
 import geotrellis.raster.reproject._
 import geotrellis.raster.resample._
 import geotrellis.proj4._
-import geotrellis.raster.io.geotiff.{MultibandGeoTiff, GeoTiffMultibandTile}
+import geotrellis.raster.io.geotiff.{AutoHigherResolution, GeoTiff, GeoTiffMultibandTile, MultibandGeoTiff, OverviewStrategy}
 import geotrellis.raster.io.geotiff.reader.GeoTiffReader
 
-import cats.implicits._
-
-class GeoTiffReprojectRasterSource(
-  val uri: String,
-  val crs: CRS,
-  val options: Reproject.Options = Reproject.Options.DEFAULT
+case class GeoTiffReprojectRasterSource(
+  uri: String,
+  crs: CRS,
+  reprojectOptions: Reproject.Options = Reproject.Options.DEFAULT,
+  strategy: OverviewStrategy = AutoHigherResolution
 ) extends RasterSource { self =>
-  def resampleMethod: Option[ResampleMethod] = options.method.some
+  def resampleMethod: Option[ResampleMethod] = Some(reprojectOptions.method)
 
   @transient lazy val tiff: MultibandGeoTiff =
     GeoTiffReader.readMultiband(getByteReader(uri), streaming = true)
 
-  protected lazy val baseCRS = tiff.crs
-  protected lazy val baseRasterExtent = tiff.rasterExtent
+  protected lazy val baseCRS: CRS = tiff.crs
+  protected lazy val baseRasterExtent: RasterExtent = tiff.rasterExtent
 
   protected lazy val transform = Transform(baseCRS, crs)
   protected lazy val backTransform = Transform(crs, baseCRS)
 
-  override lazy val rasterExtent: RasterExtent = options.targetRasterExtent match {
+  override lazy val rasterExtent: RasterExtent = reprojectOptions.targetRasterExtent match {
     case Some(targetRasterExtent) =>
       targetRasterExtent
     case None =>
-      ReprojectRasterExtent(baseRasterExtent, transform, options)
+      ReprojectRasterExtent(baseRasterExtent, transform, reprojectOptions)
   }
+
+  lazy val resolutions: List[RasterExtent] =
+    rasterExtent :: tiff.overviews.map(ovr => ReprojectRasterExtent(ovr.rasterExtent, transform))
```

Comment:
Note that I dropped the `reprojectOptions` usage here. Basically they may have either target cell size or target raster extent, which would make all of these be the same value. What we're after here is the "default" behavior of trying to estimate.